timetypes/diagnostics.go

```diff
@@ -15,3 +15,14 @@ func rfc3339InvalidStringDiagnostic(value string, err error) diag.Diagnostic {
 			"Error: "+err.Error(),
 	)
 }
+
+// durationInvalidStringDiagnostic returns an error diagnostic intended to report
+// when a string is not a time duration.
+func durationInvalidStringDiagnostic(value string, err error) diag.Diagnostic {
+	return diag.NewErrorDiagnostic(
+		"Invalid time duration String Value",
+		"A string value was provided that is not valid time duration string format.\n\n"+
```

Comment:
When receiving an invalid value error diagnostic, practitioners will likely want to know what a valid format is so they can resolve the error. The `time` package errors can be fairly opaque about what is valid, such as:

```
time: invalid duration "!"
time: missing unit in duration "10"
time: unknown unit "d" in duration "10d"
```

https://pkg.go.dev/time#ParseDuration mentions what is valid as:

> A duration string is a possibly signed sequence of decimal numbers, each with optional fraction and a unit suffix, such as "300ms", "-1.5h" or "2h45m". Valid time units are "ns", "us" (or "µs"), "ms", "s", "m", "h".

It does not need to be verbatim, but we should include this information in the diagnostic, such as:

```suggestion
		"A string value was provided that is not valid time duration string format. "+
			"A duration string is a sequence of numbers, each with optional fraction and a unit suffix, such as "300ms", "-1.5h" or "2h45m". "+
			"Valid time units are "ns", "us" (or "µs"), "ms", "s", "m", "h".\n\n"+
```

Reply:
Done :heavy_check_mark: